### ISSUE TITLE
Disable "docker login" with normal password and only allow docker login with token

### ISSUE THREAD

**Issue by Martin-Weiss** [SUSE/Portus#1795]
### Description

For security reasons we want to make sure that a user never does a docker login with a real password and just uses the tokens created for the user in Portus.

Is this possible to be configured already?

**Comment by mssola** [comment]
I don't think this is possible right now, but it makes sense. This should only be applicable to docker logins, right ? (shouldn't apply to the web, otherwise how would a user sign up if that's enabled?). Moreover, unfortunately `docker login` does not show the reasoning why a login failed, so we cannot say something like "you could not log in because you need to user an authentication token" (better phrased for security reasons, of course...). So, I'm afraid that this is potentially confusing for users. How would you make it more clear for them ?

**Comment by Martin-Weiss** [comment]
>This should only be applicable for docker logins, right ?

yes - on portus the login should only be possible via LDAP / AD account credentials and on the docker login - only tokens should be valid.

> Moreover, unfortunately docker login does not show the reasoning why a login failed, so we cannot say something like "you could not log in because you need to use an authentication token" (better phrased for security reasons, of course...). So, I'm afraid that this is potentially confusing for users. How would you make it more clear for them ?

1. I believe that in portus we need a configuration setting that states "only allow docker login with tokens".
2. in the portus audit we should show failed logins
3. docker login should be enhanced to allow more granular feedback in general on why the login failed - same is true for docker pull or push.. but that might be out of scope for the portus project and is more relevant for the docker project IMO.

**Comment by mssola** [comment]
> I believe that in portus we need a configuration setting that states "only allow docker login with tokens".

:+1: 

> in the portus audit we should show failed logins

Yes. And we could possibly add a "last logins" widget somewhere in the user's profile.

> docker login should be enhanced to allow more granular feedback in general on why the login failed - same is true for docker pull or push.. but that might be out of scope for the portus project and is more relevant for the docker project IMO.

Exactly. There's nothing we can do afaik.
